internal/provider/resource_cloudflare_ruleset_test.go

```diff
@@ -2932,3 +2995,43 @@ func testAccCloudflareRulesetCacheSettingsCustomKeyEmpty(rnd, accountID, zoneID
     }
   }`, rnd, accountID, zoneID)
 }
+
+func testAccCloudflareRulesetRedirectFromList(rnd, accountID string) string {
+	// WIP! no zone id
+	return fmt.Sprintf(`
+  resource "cloudflare_list" "list-%[1]s" {
+    account_id = "%[2]s"
+    name = "redirect_list_%[1]s"
+    description = "%[1]s list description"
+    kind = "redirect"
+  }
+
+  # We need to wait for the resource to be destroyed before the list is deleted.  Otherwise we might fail
+  # with a "list is being referenced by a ruleset".
+  resource "time_sleep" "wait" {
+    depends_on = [cloudflare_list.list-%[1]s]
+  
+    destroy_duration = "5s"
+  }
+
+  resource "cloudflare_ruleset" "%[1]s" {
+	account_id  = "%[2]s"
+    name        = "%[1]s"
+    description = "%[1]s ruleset description"
+    kind        = "root"
+    phase       = "http_request_redirect"
+
+    rules {
+      action = "redirect"
+      action_parameters {
+        from_list {
+      	  name = "redirect_list_%[1]s"
```

Comment:
you can drop the `wait` and explicit `depends_on` in favour of implicit dependencies.

```suggestion
      	  name = cloudflare_list.list-%[1]s.name
```